### PR TITLE
refactor(examples): general cleanup + improvements

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Followup to #562.
Slightly refactors and generally cleans up all examples, adding module docstrings to every file, and clarifying/improving comments.
Also fixes two examples not working at all due to missing message content intent.

~~The one thing that's still inconsistent is that some examples print a `Logged in as {bot.user} (ID: {bot.user.id})` on startup, while others don't - not sure if those should just be removed altogether, or if every example should show it.~~

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by onerandomusername** [comment]
Merge is blocked by a needed snakebot update

**Comment by shiftinv** [comment]
> Merge is blocked by a needed snakebot update

Should be fixed as of https://github.com/onerandomusername/snakebot/pull/69.

**Comment by onerandomusername** [comment]
This is still partially blocked; that snakebot pr only fixed the instant issue but doesn't solve the underlying issue.

**Comment by onerandomusername** [comment]
@shiftinv just to clarify, is this pr otherwise done, just waiting on snakebot?

**Comment by shiftinv** [comment]
> @shiftinv just to clarify, is this pr otherwise done, just waiting on snakebot?

Yup, other than that it's done, except for the logging inconsistency mentioned in the description which I'm undecided about.

**Comment by onerandomusername** [comment]
Snakebot update has been made and deployed as of commit 628ec2e1941895a8ee9b285ac80afa49dac6c2fd

**Review by onerandomusername — CHANGES_REQUESTED** [review]
Thoughts on gating every run() method with an `if __name__ == "__main__":`?

**Comment by shiftinv** [comment]
> Thoughts on gating every run() method with an `if __name__ == "__main__":`?

Fair, but I'd rather keep examples without boilerplate code as much as possible. Gating with `__name__ == "__main__"` is definitely good practice, I don't think it's needed in these examples though.

**Comment by onerandomusername** [comment]
> > Thoughts on gating every run() method with an `if __name__ == "__main__":`?
> 
> Fair, but I'd rather keep examples without boilerplate code as much as possible. Gating with `__name__ == "__main__"` is definitely good practice, I don't think it's needed in these examples though.

I think it should be done, as these examples are supposed to be the best practices

File: examples/converters.py [hunk 1]
```diff
@@ -1,6 +1,11 @@
 # SPDX-License-Identifier: MIT
 
-# This example requires the 'members' privileged intent to use the Member converter.
+"""
+An example showcasing different builtin converter types.
+"""
+
+# A list of all available converter types can be found at
+# https://docs.disnake.dev/en/latest/ext/commands/commands.html#discord-converters.
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply examples/converters.py:8]: There's a third one in this file I couldn't comment on.

File: examples/interactions/injections.py [hunk 2]
```diff
@@ -12,9 +12,8 @@
 
 bot = commands.Bot(command_prefix=commands.when_mentioned)
 
-# Instead of repeating boiler-plate code you may use injections
```
Comment (onerandomusername) [inline examples/interactions/injections.py:15]: This example throws an error when attempting to run it.

File: examples/interactions/localizations.py [hunk 3]
```diff
@@ -16,7 +16,6 @@
```
Comment (onerandomusername) [inline examples/interactions/localizations.py:16]: This example throws an error when attempting to run it.

File: examples/converters.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shiftinv) [reply examples/converters.py]: https://github.com/DisnakeDev/disnake/pull/716/commits/be4573033dd2fef43c5b2817d349ad0eb338cbaf

File: examples/converters.py [hunk 4]
```diff
@@ -19,28 +24,23 @@
...
+    # during command invocation, we will attempt to convert
     # the value passed as `user` to a `disnake.User` instance.
+
     # The documentation notes what can be converted, in the case of `disnake.User`
     # you pass an ID, mention or username (discrim optional)
-    # E.g. 80088516616269824, @Danny or Danny#0007
-
-    # NOTE: typehinting acts as a converter within the `commands` framework only.
-    # In standard Python, it is use for documentation and IDE assistance purposes.
+    # e.g. 206758669896729152, @woo or woo#9876
```
[the remark this answers is not in the document]
Comment (shiftinv) [reply examples/converters.py]: https://github.com/DisnakeDev/disnake/pull/716/commits/987a63c30fdb180373535d38e8ba1d0890d5b2c6

File: examples/interactions/injections.py (hunk 2 above)
Comment (shiftinv) [reply examples/interactions/injections.py:15]: #770

File: examples/interactions/localizations.py (hunk 3 above)
Comment (shiftinv) [reply examples/interactions/localizations.py:16]: can't reproduce, could you provide a stacktrace?

File: examples/basic_voice.py [hunk 5]
```diff
@@ -1,9 +1,14 @@
 # SPDX-License-Identifier: MIT
```
Comment (onerandomusername) [inline examples/basic_voice.py:1]: this example doesn't work

**Review by onerandomusername — CHANGES_REQUESTED** [review]

File: examples/interactions/localizations.py (hunk 3 above)
Comment (onerandomusername) [reply examples/interactions/localizations.py:16]: The example is not clear that a locale/de.json (or other supported language) is required to exist.

Comment (shiftinv) [reply examples/interactions/localizations.py:16]: https://github.com/DisnakeDev/disnake/pull/716/commits/c6eac06b1ff0db32437848fb55febef84cc1778d

File: examples/basic_voice.py (hunk 5 above)
Comment (shiftinv) [reply examples/basic_voice.py:1]: https://github.com/DisnakeDev/disnake/pull/716/commits/1f0c71acdc017eb27114d803a365424443975fbd

**Review by onerandomusername — APPROVED** [review]

**Pull request merged**